### PR TITLE
proxy: opt-out from SNAT for L7 + Tunnel for some scenarios

### PR DESCRIPTION
**Pull request by jschwinger233**
Currently the L7 proxy performs SNAT for traffic when tunnel routing is enabled, even for cluster-internal traffic. This prevents cilium_host from detecting pod-level traffic, and we thus can't apply features.

Modify SupportsOriginalSourceAddr(), so that the proxy doesn't SNAT such traffic when some conditions are met.

```release-note
Fixes proxy issues by opting out from SNAT for L7 + Tunnel.
```

### CODE REVIEW

File: pkg/datapath/iptables/iptables.go [hunk 1]
```diff
@@ -598,7 +600,7 @@ func (m *Manager) installStaticProxyRules() error {
 		if err := ip4tables.runProg([]string{
 			"-t", "filter",
 			"-A", ciliumOutputChain,
-			"-m", "mark", "--mark", matchProxyReply,
+			"-m", "mark", "--mark", matchFromProxy,
 			"-m", "comment", "--comment", "cilium: ACCEPT for proxy return traffic",
```
Comment (julianwiedmann) [inline pkg/datapath/iptables/iptables.go]: Just a cosmetic update:

```suggestion
			"-m", "comment", "--comment", "cilium: ACCEPT for proxy traffic",
```

there's also a comment on top that needs `s/proxy return/proxy`.

Same for for IPv6 below :).

**Review by julianwiedmann — CHANGES_REQUESTED** [review]
One cosmetic update below, otherwise looks good :)

**Review by julianwiedmann — APPROVED** [review]
ty!

**Review by jrajahalme — APPROVED** [review]
The comment above the change should be updated to describe why original source addressing needs to be used with IPsec.

**Comment by jrajahalme** [comment]
Comment on the PR description/commit message: While it may look like it, there is no SNAT involved with the L7 proxy. The forwarded connection is merely a new one originating from the host networking namespace.

**Pull request merged**